File: js/0089.js
/**
 * @param {number} n
 * @return {number[]}
 */
var grayCode = function(n) {
  var ans=new Array()
  for(var i=0;i<(1<<n);i++){
    ans.push(i^(i>>1))
  }
  return ans
};

File: rust/src/bin/0089.rs
struct Solution;

impl Solution {
    pub fn gray_code(n: i32) -> Vec<i32> {
        let mut ans = vec!();
        for i in 0..(1 << n) {
            ans.push(i ^ (i >> 1));
        }
        ans
    }
}

fn main() {
    let ans = Solution::gray_code(3);
    println!("{:?}", ans);
}